src/NUnitFramework/framework/Internal/TestExecutionContext.cs

```diff
@@ -493,6 +502,28 @@ public void AddFormatter(ValueFormatterFactory formatterFactory)
             CurrentValueFormatter = formatterFactory(CurrentValueFormatter);
         }
 
+        /// <summary>
+        /// Attach a file to the current test result
+        /// </summary>
+        /// <param name="filepath">Filepath to attachment. Can be absolute, or relative to working directory.</param>
+        /// <param name="description">Optional description of attachment</param>
+        public void AddTestAttachment(string filepath, string description = null)
+        {
+            Guard.ArgumentNotNull(filepath, nameof(filepath));
+            Guard.ArgumentValid(filepath.IndexOfAny(Path.GetInvalidPathChars()) == -1, 
+                $"Test attachment filepath contains invalid path characters. {filepath}", nameof(filepath));
+
+            if (!Path.IsPathRooted(filepath))
+                filepath = Path.Combine(TestContext.CurrentContext.WorkDirectory, filepath);
+
```

Comment:
Actually `Path.GetFullPath()` would be incorrect. We promise to store all output files relative to our WorkDirectory, which is __not__ the current directory. The choice of name for the WorkDirectory, may have been a bad one, since it sounds a bit like the working directory, i.e. cwd, but I believe we are stuck with it.

A more basic question... why do we want to make this absolute at all? Why not just use what the user gives us?